docs/workflow/trimming/feature-switches.md

```diff
@@ -27,6 +27,7 @@ configurations but their defaults might vary as any SDK can set the defaults dif
 | MetadataUpdaterSupport | System.Reflection.Metadata.MetadataUpdater.IsSupported | Metadata update related code to be trimmed when set to false |
 | _EnableConsumingManagedCodeFromNativeHosting | System.Runtime.InteropServices.EnableConsumingManagedCodeFromNativeHosting | Getting a managed function from native hosting is disabled when set to false and related functionality can be trimmed. |
 | VerifyDependencyInjectionOpenGenericServiceTrimmability | Microsoft.Extensions.DependencyInjection.VerifyOpenGenericServiceTrimmability | When set to true, DependencyInjection will verify trimming annotations applied to open generic services are correct |
+| DisableDependencyInjectionDynamicEngine | Microsoft.Extensions.DependencyInjection.DisableDynamicEngine | When set to true, DependencyInjection will avoid using System.Reflection.Emit when realizing services |
```

Comment:
I wonder if a more universal "AvoidEmitForPerformance" switch makes sense. That would also be checked in the couple places where the undocumented `ForceInterpretedInvoke` is used (which was originally added for testing).

Reply:
Yes, I think it makes sense to rework https://github.com/xamarin/xamarin-android/pull/7972 so it sets both feature switches on a name similar to `$(AvoidEmitForPerformance)`.

We would try to ship this in .NET 8 RC 2 if possible.